invoice/src/main/java/org/killbill/billing/invoice/api/svcs/DefaultInvoiceInternalApi.java

```diff
@@ -162,24 +157,20 @@ public InvoicePayment recordChargebackReversal(final UUID paymentId, final UUID
     @Override
     public Map<UUID, BigDecimal> validateInvoiceItemAdjustments(final UUID paymentId, final Map<UUID, BigDecimal> idWithAmount, final InternalTenantContext context) throws InvoiceApiException {
         // We want to validate that only refund with invoice *item* adjustments are allowed (as opposed to refund with invoice adjustment)
-        if (idWithAmount.size() == 0) {
+        if (idWithAmount.isEmpty()) {
             throw new InvoiceApiException(ErrorCode.INVOICE_ITEMS_ADJUSTMENT_MISSING);
         }
         final InvoicePayment invoicePayment = getInvoicePayment(paymentId, InvoicePaymentType.ATTEMPT, context);
         return dao.computeItemAdjustments(invoicePayment.getInvoiceId().toString(), idWithAmount, context);
     }
 
     private InvoicePayment getInvoicePayment(final UUID paymentId, final InvoicePaymentType type, final InternalTenantContext context) throws InvoiceApiException {
-
         final List<InvoicePaymentModelDao> invoicePayments = dao.getInvoicePaymentsByPaymentId(paymentId, context);
-        final InvoicePaymentModelDao resultOrNull = Iterables.tryFind(invoicePayments, new Predicate<InvoicePaymentModelDao>() {
-            @Override
-            public boolean apply(final InvoicePaymentModelDao input) {
-                return input.getType() == type &&
-                       input.getSuccess();
-            }
-        }).orNull();
-        return resultOrNull != null ? new DefaultInvoicePayment(resultOrNull) : null;
+        return invoicePayments.stream()
+                .filter(input -> input.getType() == type && input.getSuccess())
+                .findFirst()
+                .map(DefaultInvoicePayment::new)
```

Comment:
Is there a reason to recreate a `DefaultInvoicePayment` - it seems like the previous code would just filter based on predicate and now we filter and recreate ?

Reply:
> .... it seems like the previous code would just filter based on predicate ....

[tryFind()](https://github.com/killbill/killbill/blob/work-for-release-0.23.x/invoice/src/main/java/org/killbill/billing/invoice/api/svcs/DefaultInvoiceInternalApi.java#L177) on previous code state that: if criteria in filter found, it will create `new DefaultInvoicePayment()`, but if nothing found, it will return null. 

The current code doing exactly same. `findFirst()` return `java.util.Optional`, so `.map(DefaultInvoicePayment::new)` actually `Optional` method, not part of stream anymore. `TestDefaultInvoiceInternalApiUnit` add to test this.

Reply:
Ok, got it, I had missed the conversion `InvoicePaymentModelDao` -> `DefaultInvoicePayment`